src/Storages/Statistics/StatisticsMinMax.h

```diff
@@ -0,0 +1,31 @@
+#include <Storages/Statistics/Statistics.h>
+
+
+namespace DB
+{
+
+class StatisticsMinMax : public IStatistics
+{
+public:
+    StatisticsMinMax(const SingleStatisticsDescription & stat_, const DataTypePtr & data_type_);
+    ~StatisticsMinMax() override = default;
+
+    Float64 estimateLess(const Field & val) const override;
+
+    void update(const ColumnPtr & column) override;
+
+    void serialize(WriteBuffer & buf) override;
+    void deserialize(ReadBuffer & buf) override;
+
+private:
+    Float64 min;
+    Float64 max;
+    Float64 row_count;
```

Comment:
1. yes

Reply:
Sounds generally good to me. And yes, let's do that in a separate PR.

Before doing refactoring, it would be appreciated to go over the tests again (https://github.com/ClickHouse/ClickHouse/pull/67013#discussion_r1703151496) and make them as systematic as possible.

Reply:
@JackyWoo the idea is good. I am not sure how the details is implemented. Let's do it and refine it if needed